src/lib/chainNode.ts

```diff
@@ -174,57 +174,44 @@ export default class ChainNode {
   async submitRunProcess(
     extrinsic: SubmittableExtrinsic<'promise', SubmittableResult>,
     transactionDbUpdate: (state: TransactionState) => Promise<void>
-  ): Promise<number[]> {
-    this.logger.debug('Submitting Transaction %j', extrinsic.hash.toHex())
-    return new Promise((resolve, reject) => {
-      let unsub: () => void
-      extrinsic
-        .send((result: SubmittableResult) => {
-          this.logger.debug('result.status %s', JSON.stringify(result.status))
-
-          const { dispatchError, status } = result
-
-          if (dispatchError) {
-            this.logger.warn('dispatch error %s', dispatchError)
-            transactionDbUpdate('failed')
-            if (dispatchError.isModule) {
-              const decoded = this.api.registry.findMetaError(dispatchError.asModule)
-              reject(new Error(`Node dispatch error: ${decoded.name}`))
-            } else {
-              reject(new Error(`Unknown node dispatch error: ${dispatchError}`))
-            }
-            unsub()
-            return
-          }
+  ): Promise<void> {
+    try {
+      this.logger.debug('Submitting Transaction %j', extrinsic.hash.toHex())
+      const unsub: () => void = await extrinsic.send((result: SubmittableResult): void => {
+        this.logger.debug('result.status %s', JSON.stringify(result.status))
 
-          if (status.isInBlock) {
-            transactionDbUpdate('inBlock')
+        const { dispatchError, status } = result
+
+        if (dispatchError) {
+          this.logger.warn('dispatch error %s', dispatchError)
+          transactionDbUpdate('failed')
```

Comment:
need to unsub if there's a dispatch error

Reply:
oops, you did mention in the chat, it has been added